mmpose/datasets/pipelines/mesh_transform.py

```diff
@@ -0,0 +1,370 @@
+        results['joints_3d_visible'] = joints_3d_visible
+        results['pose'] = pose
+        results['center'] = center
+
+        if 'iuv' in results.keys():
+            iuv = results['iuv']
+            if np.random.rand() <= self.flip_prob and iuv is not None:
+                iuv = flip_iuv(iuv, results['ann_info']['uv_type'])
+            results['iuv'] = iuv
+
+        return results
+
+
+@PIPELINES.register_module()
+class MeshGetRandomScaleRotation():
+    """Data augmentation with random scaling & rotating.
+
+    Required key: 'scale'. Modifies key: 'scale' and 'rotation'.
+
+    Args:
+        rot_factor (int): Rotating to ``[-2*rot_factor, 2*rot_factor]``.
+        scale_factor (float): Scaling to ``[1-scale_factor, 1+scale_factor]``.
+        rot_prob (float): Probability of random rotation.
+    """
+
+    def __init__(self, rot_factor=30, scale_factor=0.25, rot_prob=0.6):
+        self.rot_factor = rot_factor
+        self.scale_factor = scale_factor
+        self.rot_prob = rot_prob
+
+    def __call__(self, results):
+        """Perform data augmentation with random scaling & rotating."""
+        s = results['scale']
+
+        sf = self.scale_factor
+        rf = self.rot_factor
+
+        s_factor = np.clip(np.random.randn() * sf + 1, 1 - sf, 1 + sf)
+        s = s * s_factor
+
+        r_factor = np.clip(np.random.randn() * rf, -rf * 2, rf * 2)
+        r = r_factor if np.random.rand() <= self.rot_prob else 0
+
+        results['scale'] = s
+        results['rotation'] = r
+
+        return results
+
+
+@PIPELINES.register_module()
+class MeshAffine():
+    """Affine transform the image to make input.
+
+    Required keys:'img', 'joints_3d', 'joints_3d_visible', 'ann_info','scale',
+    'rotation' and 'center'. Modified keys:'img', 'joints_3d', and
+    'joints_3d_visible'.
+
+    Required keys: 'img', 'joints_2d','joints_2d_visible', 'joints_3d',
+     'joints_3d_visible', 'pose', 'iuv', 'ann_info','scale',
+     'rotation' and 'center'.
+    Modifies key: 'img', 'joints_2d','joints_2d_visible', 'joints_3d',
+     'pose', 'iuv'.
+    """
+
+    def __call__(self, results):
+        image_size = results['ann_info']['image_size']
+
+        img = results['img']
+        joints_2d = results['joints_2d']
+        joints_2d_visible = results['joints_2d_visible']
+        joints_3d = results['joints_3d']
+        pose = results['pose']
+
+        c = results['center']
+        s = results['scale']
+        r = results['rotation']
+        trans = get_affine_transform(c, s, r, image_size)
+
+        img = cv2.warpAffine(
+            img,
+            trans, (int(image_size[0]), int(image_size[1])),
+            flags=cv2.INTER_LINEAR)
+
+        for i in range(results['ann_info']['num_joints']):
+            if joints_2d_visible[i, 0] > 0.0:
+                joints_2d[i] = affine_transform(joints_2d[i], trans)
+
+        joints_3d = rotate_joints_3d(joints_3d, r)
+        pose = rot_smpl_pose(pose, r)
```

Comment:
if `rot_smpl_pose` is an internal function, then make it start with `_`

Reply:
ok